keanu-python/build.gradle

```diff
@@ -101,40 +100,41 @@ task generateDocumentation {
     }
 }
 
-def keanuVertexGeneratedPath
-def keanuVertexInitPath
-
-task populateExclusionPaths {
-    def keanuVertexPath = Paths.get("keanu", "vertex")
-    if(isFamily(FAMILY_WINDOWS)) {
-        keanuVertexPath = Paths.get(".").resolve(keanuVertexPath)
-    }
-    keanuVertexGeneratedPath = keanuVertexPath.resolve("generated.py")
-    keanuVertexInitPath = keanuVertexPath.resolve("__init__.py")
-}
+List<String> formatExcludes = [
+        '--exclude', project.file('keanu/vertex/generated.py'),
+        '--exclude', project.file('keanu/vertex/__init__.py'),
```

Comment:
I would expect this to fail on Windows because (1) the path separator needs to be `\` (2) For some reason it only works on Windows when the path is prefixed by `.\` prefix

Reply:
Yeah I'd hoped `.file` would somehow magically deal with this for us but I'm guessing it's an issue with `yapf`? Should it break the CI builds?

Reply:
Seems like  the `appveyor` build should trigger issues? If not, perhaps we should ensure it does?

Reply:
I'm surprised it's passing on AppVeyor. @wesleyw72 had a lot of grief with this when he did the original implementation.

Reply:
Or maybe it doesn't fail, it just ignores the `--exclude`?

Reply:
Oh, I see what's happening now - you're actually not executing any command in `formatCheck` and `formatApply`!

The `addAll` returns `true`, so all you are actually just executing `commandLine(true)` which trivially succeeds.

So instead you want something like:

        def command = ['pipenv', 'run', 'yapf', '--diff', '--recursive', '--parallel', project.rootDir]
        command.addAll(formatExcludes)
        exec {
            commandLine(command)
        }

Reply:
On the plus side - `project.file` does format it correctly for Windows as you predicted (I checked on a laptop). But it actually fails for a different reason, which is that it is running (and failing) against `keanu\docs`. This is because you are pointing yapf at `project.rootDir` (i.e. `keanu`) when it should be `.` (i.e. `keanu\keanu-python`)

Reply:
Yep just realised that. I've made a change now to use `.yapfignore` instead. We'll see how that goes.

Reply:
👍

Reply:
Nice! I wanted to do it with an ignore file when I first did it but they didn't have the functionality then. Good spot that they've added this!